### ISSUE TITLE
Ruby version support

### ISSUE THREAD

**Issue by ianks** [rake-compiler/rake-compiler-dock#77]
Currently, RCD supports all the way down to Ruby 2.4 which has been EOL for almost 3 years.

Not only is there a maintenance burden from supporting more versions, it makes certain refactors harder to do.

I propose come up with a policy for which Ruby versions RCD will support, and how that will affect RCDs semver. The obvious proposal would be to support stable rubies (as of now 2.7+), and release a major version of RCD whenever a Ruby version is EOL.

We can also adjust this scheme to make migration easier — i.e. support the most recent EOL Ruby version (as of now 2.6) via opt-in if it’s included in `RUBY_CC_VERSION`.

The goal here is to agree on a support policy going forward, since it’s untenable to support all rubies forever.

Thoughts?

**Comment by flavorjones** [comment]
Wearing my Nokogiri maintainer hat, I try to support the 4 most recent Ruby minors at any time. A policy like this would be fine for me, but I think we should solicit more opinions from other users before committing to anything.

**Comment by copiousfreetime** [comment]
I have [amalgalite](https://rubygems.org/gems/amalgalite/) - which doesn't have a lot of usage, but its still there (and yes, it needs a new release). I think supporting the most recent EOL version (either as an extra commandline, or by default) and all minors going forward with a major version release of RCD after each EOL sounds like a reasonable policy.

**Comment by larskanis** [comment]
RCD is meant to support the release process of other gems with possibly a longer support time frame. That's why I tried to support 2 or 3 EOL releases as long as there is no relevant maintenance burden. And that wasn't the case so far. It only took a little bit more building time on my end and approx 25 MByte per release uncompressed.

**Comment by flavorjones** [comment]
Bumping this thread because I'm struggling to make Ruby 3.0.x work for x > 2. See #134, #135, #136 for the gory details.

**Comment by flavorjones** [comment]
Bumping this again because of the build issues raised in https://github.com/mudge/re2/pull/167 which seem to be due to conflicts between the fake.rb rbconfig and the actual library being loaded.

I am more convinced every day that this tool should not encourage precompiling EOL versions of Ruby.

**Comment by rkh** [comment]
Agreed. Moving most of my gems to respect Ruby EOL has made my life much easier. Maintainers can still choose to support EOL versions, but they should not ship precompiled gems for them. This would also help with gem size, given that you can't publish separate gems for separate Ruby versions, only platforms.
